pkg/execution/driver/httpdriver/httpdriver.go

```diff
@@ -242,21 +242,24 @@ func do(ctx context.Context, c *http.Client, r Request) (*response, error) {
 	}
 
 	if err != nil && !errors.Is(err, io.EOF) {
-		if err != nil {
-			if urlErr, ok := err.(*url.Error); ok && urlErr.Err == context.DeadlineExceeded {
-				// This timed out.
-				return nil, context.DeadlineExceeded
-			}
-			if errors.Is(err, syscall.EPIPE) {
-				return nil, fmt.Errorf("Your server closed the request before finishing.")
-			}
-			if errors.Is(err, syscall.ECONNRESET) {
-				return nil, fmt.Errorf("Your server reset the request connection.")
-			}
+		if urlErr, ok := err.(*url.Error); ok && urlErr.Err == context.DeadlineExceeded {
+			// This timed out.
+			return nil, context.DeadlineExceeded
+		}
+		if errors.Is(err, syscall.EPIPE) {
+			return nil, fmt.Errorf("Your server closed the request before finishing.")
+		}
+		if errors.Is(err, syscall.ECONNRESET) {
+			return nil, fmt.Errorf("Your server reset the request connection.")
+		}
+		// Unexpected EOFs are valid and returned from servers when chunked encoding may
+		// be invalid.  Handle any other error by returning immediately.
+		if !errors.Is(err, io.ErrUnexpectedEOF) {
```

Comment:
did you mean `errors.Is` and not `!errors.Is`?
the code and the comment doesn't seem to match unless I'm misunderstanding something.